compiling/extract_and_compile.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+nthArg=$1
+
+#Uncompress and extract NthPrime.tgz into current directory
+#x - extracts contents of given file
+#f - name of file to uncompress
+#z - uncompress tarball via gzip
+tar xfz NthPrime.tgz
+
+#cd into newly created NthPrime directory
+cd NthPrime || exit
+
+#Compile NthPrime program using gcc (preprocesses, compiles, and assembles) 
+gcc ./*.c -o NthPrime
```

Comment:
What you have is correct, but I'd probably removed the `./` since it's not necessary in this context.

```suggestion
gcc *.c -o NthPrime
```